.github/workflows/rust.yml

```diff
@@ -3,6 +3,39 @@ on: [push]
 name: Rust-CI
 
 jobs:
+  rustfmt:
+    name: rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+
+      - name: install nightly
```

Comment:
As noted in the PR description, this could be replaced by a `runs-on:` item that specify a rust-nightly container no?

Reply:
Right, it can be replaced with the item `container: rustlang/rust:nightly`.
The `runs-on:` item only describes the the type of the VM. It is a bit confusing. If you only specify the `runs-on:` item, the steps will run on the VM. If you additionally specify a `container` item, the steps will run in that docker container.
The `container` feature makes the workflow easier to read but it is also a bit slower 10-20 sec. Shall we use it anyway?

Reply:
But we would also shave time off by skipping the install step no?

Reply:
Right, but the installation step takes less time.
I just tested it again. The `Initialize containers` step takes 39 sec and the `install nightly` 17 sec.

Reply:
I see. Well since you've been working on improving the build time with the cache I think it makes sense to follow the same road and optimize for time as well.